src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -192,63 +162,146 @@ class instance_scarlet_monastery : public InstanceMapScript
     };
 };
 
+class at_scarlet_monastery_cathedral_entrance : public OnlyOnceAreaTriggerScript
+{
+public:
+    at_scarlet_monastery_cathedral_entrance() : OnlyOnceAreaTriggerScript("at_scarlet_monastery_cathedral_entrance") {}
+
+    bool _OnTrigger(Player* player, AreaTrigger const* /*trigger*/) override
+    {
+        if (InstanceScript* instance = player->GetInstanceScript())
+        {
+            if (player->HasAura(AURA_OF_ASHBRINGER))
+            {
+                Creature* commanderMograine = ObjectAccessor::GetCreature(*player, instance->GetGuidData(DATA_MOGRAINE));
+                if (commanderMograine)
+                {
+                    std::list<Creature*> ScarletList;
+                    player->GetCreatureListWithEntryInGrid(ScarletList, NPC_SCARLET_MYRIDON, 4000.0f);
+                    player->GetCreatureListWithEntryInGrid(ScarletList, NPC_SCARLET_DEFENDER, 4000.0f);
+                    player->GetCreatureListWithEntryInGrid(ScarletList, NPC_SCARLET_CENTURION, 4000.0f);
+                    player->GetCreatureListWithEntryInGrid(ScarletList, NPC_SCARLET_SORCERER, 4000.0f);
+                    player->GetCreatureListWithEntryInGrid(ScarletList, NPC_SCARLET_WIZARD, 4000.0f);
+                    player->GetCreatureListWithEntryInGrid(ScarletList, NPC_SCARLET_ABBOT, 4000.0f);
+                    player->GetCreatureListWithEntryInGrid(ScarletList, NPC_SCARLET_MONK, 4000.0f);
+                    player->GetCreatureListWithEntryInGrid(ScarletList, NPC_SCARLET_CHAMPION, 4000.0f);
+                    player->GetCreatureListWithEntryInGrid(ScarletList, NPC_SCARLET_CHAPLAIN, 4000.0f);
+                    player->GetCreatureListWithEntryInGrid(ScarletList, NPC_FAIRBANKS, 4000.0f);
```

Comment:
do we really need to search this big..:?

Reply:
This is the range in the old code, and I guess the reason the range is so large : triggering the Ashbringer event makes the wizards and monks in the library friendly as well. The library is also in the same map

Reply:
I tested it, and even 4,000 yards wouldn't make a wizard or monk in the library friendly

Reply:
I will try to find someone on Classic to test, but afaik the blade has no effect outside of the cathedral wing.

Reply:
> [...] afaik the blade has no effect outside of the cathedral wing.

I was mistaken. The blade has no effect in the graveyard, but in the library, armory, and cathedral it has the famous effect on the Scarlets.

Reply:
I've changed the way and now I don't use range search to modify factions. The previous range of use should also work. Now I'm using two accounts to team up. A pre-entry to the library. An entry into the Cathedral triggers an event. Some of the library's NPCs changed factions as expected. But if I go to the library by entering the Cathedral first to trigger the event, I will find that the NPC faction inside has not changed. There is also the first to enter the library. Go to the cathedral to trigger the event. Then go back to the library to find that the faction has changed. It seems that the NPCs in the library and armory are not loaded when entering the cathedral to trigger the event. Although they are all on the same map

Reply:
Access to the library and armory. Going to the Cathedral again triggers the event and it will change factions as expected Probably something to do with the world.conf I remember there was an option to load all map data. I don't have enough memory on my computer to open that option test